src/main/java/com/emc/object/util/RestUtil.java

```diff
@@ -259,11 +259,10 @@ public static URI buildUri(String scheme, String host, int port, String path, St
         // workaround for https://bugs.openjdk.java.net/browse/JDK-8037396
         uriString = uriString.replace("[", "%5B").replace("]", "%5D");
 
-        // replace double-slash with /%2f (workaround for apache client)
-        if (path != null && path.length() > 2 && path.charAt(0) == '/' && path.charAt(1) == '/') {
-            int doubleSlashIndex = uriString.indexOf("//");
-            if (scheme != null) doubleSlashIndex = uriString.indexOf("//", doubleSlashIndex + 2);
-            uriString = uriString.substring(0, doubleSlashIndex) + "/%2F" + uriString.substring(doubleSlashIndex + 2);
+        // replace double-slash with /%2f (otherwise apache client will normalize it to one slash)
+        if (path.contains("//")) {
+            if (scheme != null) uriString = uriString.substring(0, 8) + uriString.substring(8).replaceAll("//", "/%2F");
```

Comment:
Is it possible that path could be null by an unfriendly usage? 
Will it work to process path before creating uri?